### ISSUE TITLE
AppImage for ARM systems

### ISSUE THREAD

**Issue by jschwartzenberg** [KDAB/hotspot#228]
**Is your feature request related to a problem? Please describe.**
I would like to analyze a problem on my ASUS Tinkerboard which runs Ubuntu 18.04. On my other system I already saw that Hotspot requires a newer version of Qt than what's in the latest Ubuntu LTS. When I load the `perf.data` I generated on the Tinkerboard in Hotspot on that system, it misses all the libraries (obviously), so I'd like to be able to easily run Hotspot on the Tinkerboard itself.

**Describe the solution you'd like**
Have an AppImage for ARM available.

**Describe alternatives you've considered**
Allow compilation against Qt 5.9 (instead of requiring 5.10).

**Additional context**
Maybe there are more quick ways to solve this that I did not consider.

**Comment by jschwartzenberg** [comment]
For reference, if you run into a Ubuntu 18.04 with an unsupported architecture (or there's another reason to recompile), 6bf4e78e32a6d5c68f0e19fdce01d98d8c9dd2da is the last compileable commit on Ubuntu 18.04 (with Kubuntu backports enabled). And don't forget to update the submodules after checking out this revision.

**Comment by milianw** [comment]
You don't need the GUI on the tinkerboard - just continue compiling. Afaik the Qt dependency is optional and it should figure out automatically that it's not fulfilled. Continuing the compile step you should still get the injectable library which you can use to record the data. Then copy the data to another machine where you can run the AppImage

**Comment by jschwartzenberg** [comment]
That's true when I have a second AMD64 system on hand, but being able to do without would be neat. For now I used the older commit ID which did the trick for what I needed to do.

**Comment by milianw** [comment]
right, I see that this will become more useful in the future, now that more and more laptops etc. with ARM get released that could be used for development purposes. We'll have to find a way to either cross compile for arm on the CI, or find a CI that runs on ARM directly to build the AppImages...

**Comment by jintiangoodday1** [comment]
It would be more easily to do cross-compiling if hotspot removes all kde dependency and use only Qt API.

**Comment by milianw** [comment]
@jintiangoodday1 not going to happen.

**Comment by NN---** [comment]
Any update on this ?

**Comment by fifbroman8** [comment]
I have static link version for arm64 https://github.com/fifbroman8/hotspot-bin/releases/tag/v4.0.0
But I don't have arm64 device to test it yet.
